tripy/tripy/frontend/utils.py

```diff
@@ -405,6 +405,76 @@ def convert_nontensor_arg(arg, list_index=None):
 
     return impl
 
+def wraps_to_flat_ir_to_func(cls):
```

Comment:
I don't know if this decorator is the approach we want to take. The problem is that it doesn't generalize to helpers like broadcast, which is probably the main motivation for this change. Is there a way we can decouple this from the class and make it instead apply to the function/method directly? Maybe we can have a requirement that the function returns the outputs (that can be automated for `to_flat_ir` and then we can trace back the entire FlatIR function body from there

A very rough sketch:
```py
def make_function(func):
    @functools.wraps(func)
    def wrapped(*args, **kwargs, flat_ir=None):
        outputs = func(*args, **kwargs)
        
        # Trace back the inputs from the outputs and then build a FlatIR function
        flat_ir_function = FlatIRFunction(...)

        return outputs
    return wrapped
```